### PR TITLE
Read configuration from HEAD

### LINKED ISSUE #134: Config is loaded from master
This really annoys me. Going to fix it.

### PR DESCRIPTION
**Pull request by rmccue**
Reads the hmlinter config from the commit being checked rather than the main branch, allowing it to follow regular PR change processes. I had hoped probot would fix this, but https://github.com/probot/probot/pull/1188 indicates they don't want to, so let's build our own config function.

Fixes #134.

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -15,6 +18,37 @@ const DEFAULT_CONFIG = {
...
+			foo: 'bar',
+		} ),
+		ref: head,
+	};
+
+	try {
+		const res = await context.github.repos.getContent( params );
+		const config = yaml.safeLoad( Buffer.from( res.data.content, 'base64' ).toString() ) || {};
+		return {
+			...defaultConfig,
```
Comment (mikeselander) [inline src/config.js:44]: `defaultConfig` doesn't appear to be defined, so I don't think we're actually merging our config in with the repo's properly.

Comment (rmccue) [reply src/config.js:44]: Whoops, you're right, forgot to update this reference when I did the other one 🤦

File: src/config.js [hunk 2]
```diff
@@ -15,6 +18,37 @@ const DEFAULT_CONFIG = {
...
+ *
+ * @param context Context from Probot
+ * @param head SHA of the head commit we're running against.
+ * @return Configuration object read from the file
+ */
+module.exports = async ( context, head ) => {
+	const params = {
+		...context.repo( {
+			path: path.posix.join( '.github', FILENAME ),
+			foo: 'bar',
```
Comment (mikeselander) [inline src/config.js:35]: Is this an accidental inclusion or does it serve a purpose?

Comment (rmccue) [reply src/config.js:35]: wat, how did I do that

Comment (rmccue) [inline src/config.js:35]:
```suggestion
			path: path.posix.join( '.github', FILENAME ),
```

**Pull request merged**